src/opentimelineio/timeline.cpp

```diff
@@ -27,8 +27,8 @@ void Timeline::write_to(Writer& writer) const {
 
 std::vector<Track*> Timeline::video_tracks() const {
     std::vector<Track*> result;
-    for (auto c: _tracks.value->children()) {
-        if (Track* t = dynamic_cast<Track*>(c.value)) {
+    for (auto c: _tracks->children()) {
+        if (auto t = dynamic_retainer_cast<Track>(c)) {
```

Comment:
This would be a `Retainer<Track>`, a `Retainer` class templated with the `Track` type.

I'm a big fan of the `auto` keyword but it can be a bit mysterious sometimes, especially with references. I think I've read some places have coding policies against using it, but it is so convenient, especially with the STL classes. For example this:
```
auto c = _tracks->children().begin();
```
Vs:
```
std::vector<SerializableObject::Retainer<Composable>>::const_iterator c = _tracks->children().begin();
```
And if you make a typo in the above, this is the handy error message you get :)
```
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E0312	no suitable user-defined conversion from "std::_Vector_const_iterator<std::_Vector_val<std::conditional_t<true, std::_Simple_types<opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable>>, std::_Vec_iter_types<opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable>, size_t, ptrdiff_t, opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> *, const opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> *, opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> &, const opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> &>>>>" to "std::_Vector_iterator<std::_Vector_val<std::conditional_t<true, std::_Simple_types<opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable>>, std::_Vec_iter_types<opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable>, size_t, ptrdiff_t, opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> *, const opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> *, opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> &, const opentimelineio::v1_0::SerializableObject::Retainer<opentimelineio::v1_0::Composable> &>>>>" exists	opentimelineio	C:\dev\otio\OpenTimelineIO\src\opentimelineio\timeline.cpp	30	
```

Reply:
My general rule is for auto is to apply DRY.

good auto:

```cpp
auto foo = std::make_shared<Some::Inconvenient<type::NameFromHell>>()
```
auto saves us from typing/reading the same strangled types twice

bad auto

```cpp
auto arentIMysterious = someWeirdoFunction();
```